src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,321 @@
+     * If the task type does not match Duke's supporting tasks
+     * it would not create a Task.
+     *
+     * @param userInput The whole user command as String.
+     * @param taskType The task type as String
+     */
+    public void addTaskToTaskList(String userInput, String taskType) {
+        Task newTask;
+        switch (taskType.toLowerCase()){
+        case "deadline":
+            newTask = createDeadlineTask(userInput);
+            break;
+        case "event":
+            newTask = createEventTask(userInput);
+            break;
+        case "todo":
+            newTask = createTodoTask(userInput);
+            break;
+        default:
+            System.out.println("Invalid type of task given!");
+            return;
+        }
+        if (newTask != null) {
+            listOfTask.add(newTask);
+            printTaskUpdate(newTask, "added");
+            saveTaskListToFile();
+        }
+    }
+
+    /**
+     * Updates the status of the task with specified task number,
+     * to either done or not done.
+     * The function processes the whole user input as a string.
+     * The function would get the task number
+     * to be marked/unmarked from the user input.
+     * If an invalid task number is provided, no task is marked.
+     * Else, it would update the task status with the respective status provided.
+     *
+     * @param userInput The whole user command as String.
+     * @param taskStatus The new status of the task as String.
+     */
+    public void markTaskInTaskList(String userInput, String taskStatus) {
+        try {
+            int indexOfTaskInList = getTaskNumberFromInput(userInput);
+            // true if it is "mark", set to false if it is "unmark"
+            boolean isTaskDone = taskStatus.equalsIgnoreCase("mark");
+            Task taskToBeMark = getTaskFromListOfTask(indexOfTaskInList);
+            taskToBeMark.setDone(isTaskDone);
+            if (isTaskDone) {
+                System.out.println("Nice! I'v marked this task as done:");
+            } else {
+                System.out.println("Okay! I'v marked this task as not done:");
+            }
+            System.out.println(taskToBeMark);
+            saveTaskListToFile();
+        } catch (IndexOutOfBoundsException idxError) {
+            System.out.println("Invalid task number to be marked!");
+        } catch (NumberFormatException numFormError) {
+            System.out.println("Please enter a number to mark task.");
+        }
+    }
+
+    /**
+     * Deletes a task with specified task number.
+     * The function processes the whole user input as a string.
+     * The function would get the task number to be deleted from the user input.
+     * If an invalid task number is provided, no task is deleted.
+     * Else, the task would be deleted from the TaskList.
+     *
+     * @param userInput The whole user command as String.
+     */
+    public void deleteTaskInTaskList(String userInput) {
+        try {
```

Comment:
Good job overall in ensuring [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) in most methods!